File: src/main/java/com/webapp/IgniteStateService.java
package com.webapp;

import com.webapp.model.IgniteMetricsContainer;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.util.function.Tuple2;

import java.time.Duration;
import java.util.stream.Stream;

@Service
public class IgniteStateService {
    public Flux<IgniteMetricsContainer> getIgniteState() {
        Flux<Long> interval = Flux.interval(Duration.ofSeconds(1));
        Flux<IgniteMetricsContainer> igniteMetricsContainerFlux = Flux.fromStream(Stream.generate(IgniteMetricsContainer::new));
        return Flux.zip(interval, igniteMetricsContainerFlux).map(Tuple2::getT2);
    }
}

File: addressbook-main/src/main/kotlin/com/addressbook/services/IgniteStateService.kt
package com.addressbook.services

import com.addressbook.ignite.IgniteMetricsContainer
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import reactor.core.publisher.Flux
import reactor.util.function.Tuple2

import java.time.Duration
import java.util.stream.Stream

@Service
class IgniteStateService {

    @Autowired
    lateinit var igniteMetricsContainer: IgniteMetricsContainer

    fun getIgniteState(): Flux<IgniteMetricsContainer> {
        val interval = Flux.interval(Duration.ofSeconds(1))
        val igniteMetricsContainerFlux = Flux.fromStream(Stream.generate(igniteMetricsContainer::refresh))
        return Flux.zip(interval, igniteMetricsContainerFlux).map(Tuple2<Long, IgniteMetricsContainer>::getT2)
    }
}
